lib/typed_ecto_schema/ecto_type_mapper.ex

```diff
@@ -84,6 +89,18 @@ defmodule TypedEctoSchema.EctoTypeMapper do
     end
   end
 
+  defp base_type_for({:__aliases__, _, [:Ecto, :Enum]}, opts) do
+    opts
+    |> Keyword.get(:values, [])
+    |> disjunction_typespec()
+  end
+
+  defp base_type_for({:__aliases__, _, _} = ast, _opts) do
+    quote do
+      unquote(ast).t()
+    end
+  end
```

Comment:
What this does?
Is it because the type generation was only working if no aliases were being used?
Do we have any testes for that to avoid regression?

Reply:
Ah, are this what solves the compile deps problem?

Reply:
Ah, okay, thanks for the explanation via Slack.
So for further reference, this is a required step because now we are calling `TypedEctoSchema.TypeBuilder.add_field/5` with a escaped macro to remove the compile time dependency, therefore we don't necessarily have the resolved value and more often than not we have an alias, so this is required to make it work on that scenario while still not requiring a compile time dependency.

Thanks! That is a neat job!